andela/mmdp_site
# MMDP-784: Fill State and LGA Dropdown options based on Uploaded SVG map data.

**What does this PR do?**

Use the state SVG maps data to populate state and LGA dropdown information.

**Description of Task to be completed?**

Fetch state from under `statemaps` table rather than states table from the database. Use `lgaboundaries` table for the `lgas` of the a given state.

- Adds post save hook to StateMaps collection
- Adds logic to save new states after state maps are uploaded
- Still needs to add tests for the logic
- Still needs to add lga boundaries logic

**How should this be manually tested?**

- Clone this repo and cd into the folder `git clone https://github.com/andela/mmdp_site && cd mmdp_site && yarn install`
- Checkout into this branch `ft-state-lga-dropdowns-fill-mmdp-784 `
- Type `yarn run test:client`
- Run the app with the following command `yarn run start`
- Visit `http://localhost:1234/matrix/country`
- Upload a new country/national level SVG map.
- Navigate to `http://localhost:1234/dropdowns/state`
- The states data should be populated based on the country map you uploaded

**Any background context you want to provide?**

- For data integrity purposes one would need to delete their current data within the `statemaps` and 'countrymaps' collections and reupload the maps to ensure the logic for updating the `states` collection is triggered.

- LGA - Local Government Area

**What are the relevant Jira issues?**

[MMDP-784](https://viisaus.atlassian.net/browse/MMDP-784)

**Screenshots (if appropriate)**

Coming soon...
